docs/aws/audit/iamcompliance/rules/iam_customer_policies_for_blocked_kms_actions.mdx

```diff
@@ -23,6 +23,285 @@ CBP
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access IAM policies.
+
+2. Once AWS CLI is set up, you can list all the IAM policies using the following command:
+   ```
+   aws iam list-policies --scope Local
+   ```
+   This command will return a list of all IAM policies in your AWS account.
+
+3. To check the details of each policy, you can use the following command:
+   ```
+   aws iam get-policy-version --policy-arn <Policy_ARN> --version-id <Policy_Version_ID>
+   ```
+   Replace `<Policy_ARN>` and `<Policy_Version_ID>` with the ARN and version ID of the policy you want to check. This command will return the details of the specified policy version.
+
+4. Now, you need to manually inspect the policy details to check if there are any blocked KMS actions. Look for the "Effect": "Deny" statement in the policy document. If there are any KMS actions listed under this statement, it means those actions are blocked.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. **Setup AWS SDK for Python (Boto3):**
+   First, you need to set up AWS SDK for Python (Boto3) in your local environment. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   After installing boto3, you need to configure it. You can do this by creating a new session using your AWS credentials:
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       aws_session_token='SESSION_TOKEN',
+   )
+   ```
+2. **List IAM Policies:**
+   Use the `list_policies` method to retrieve all the IAM policies. This method returns a dictionary containing metadata about each IAM policy.
+   ```python
+   client = session.client('iam')
+   policies = client.list_policies(Scope='All')
+   ```
+3. **Get Policy Details:**
+   For each policy, use the `get_policy_version` method to retrieve the policy document. This method returns a dictionary containing the policy document.
+   ```python
+   for policy in policies['Policies']:
+       default_version_id = policy['DefaultVersionId']
+       policy_version = client.get_policy_version(
+           PolicyArn=policy['Arn'],
+           VersionId=default_version_id
+       )
+       policy_document = policy_version['PolicyVersion']['Document']
+   ```
+4. **Check for Blocked KMS Actions:**
+   Parse the policy document and check if any KMS actions are blocked. If a policy blocks any KMS actions, print the policy name and the blocked actions.
```

Comment:
<h2>Secret Detected: Secret Keyword</h2> <a href="https://console.cloudanix.com/code?t=secrets"> Click here for more details</a>